src/deluge/model/song/song.cpp

```diff
@@ -2915,37 +2915,25 @@ const char* Song::getScaleName(int32_t scale) {
 
 int32_t Song::cycleThroughScales() {
 	int32_t currentScale = getCurrentPresetScale();
-	if (currentScale >= NUM_PRESET_SCALES) {
-		// If we are currently in a CUSTOM scale, we can't cycle to normal scales
-		return 255;
-	}
 	int32_t newScale = currentScale + 1; // next scale
 	if (newScale >= NUM_PRESET_SCALES) {
 		// If past the last scale, start from the beginning
 		newScale = 0;
 	}
-	// Set it, it will
-	return setPresetScale(newScale);
+	int32_t result = setPresetScale(newScale);
```

Comment:
use the error enum

Reply:
The method returns the scale that it was able to set given the current conditions. It is not just errors, or is the error enum also used for success values?

Reply:
Oh ok - make a new enum class instead then. Don't just return ints with special meaning

Reply:
But what about the scales indexes? if it returns from 0 to (NUM_SCALES - 1) as valid, that is, the method could apply a known scale, but if not, there are two possible error cases. Should I return a tuple? first value is the applied preset scale and second value the error enum?

Reply:
The old code was returning 255 to signal a CUSTOM scale, because all the other parts of the code check for:

int32_t changedScale = setPresetScale(newScale)
**if (changedScale >= NUM_PRESET_SCALES) {
    // do sth with Custom scale case**
    
 I can do the refactor in all places but not sure if worth it

Reply:
Found a better way to approach it, without needing that constant to represent a special error. Commit pushed.

Reply:
Great, I like it!